### PR TITLE
Start using vale and move docs source

### PR DESCRIPTION
**Pull request by akaszynski**
This PR proposes the use of [Vale.sh](https://vale.sh/).

As of yet, we rely on humans to review the grammer and content of our documentation, but I'd like to automate this by using Vale. This PR starts by adding in Vale's configuration and aligning a single file (``user-guide/vtk_to_pyvista.rst``) with the Google style guide.

Here's an example output:
```
$ vale source/user-guide/data_model.rst 

 source/user-guide/data_model.rst
 3:1     warning  'PyVista Data Model'            Google.Headings 
                  should use sentence-style                       
                  capitalization.                                 
 7:27    error    'k. M' should have one space.   Google.Spacing  
 7:39    warning  Try to avoid using              Google.We       
                  first-person plural like                        
                  'our'.                                          
 18:35   warning  Try to avoid using              Google.We       
                  first-person plural like                        
                  'our'.                                          
 21:1    warning  'The PyVista DataSet'           Google.Headings 
                  should use sentence-style                       
                  capitalization.                                 
 28:19   warning  Try to avoid using              Google.We       
                  first-person plural like 'we'.                  
 35:6    warning  Try to avoid using              Google.We       
                  first-person plural like 'we'.                  
```

---

A few notes:

- I suggest going with the Google Developer Documentation Style Guide as it seems to strike more of a conversational tone vs. some of the other style guides. I've included it in this PR.
- This PR moves the source of the documentation to the ``source`` directory. This seems more organized to me, especially with the inclusion of additional style configuration files.
- I'm using ChatGPT to help rewrite certain sections. Originally I used it to just get rid of first person, but I find that the rewritten sections read much better than the original sections. I'm honestly not the best technical writer, but I can discrimate between *ok* and good writing, and I think we can achieve better results with this approach provided we do not lose the original content and meaning.
- We can still use codespell since it runs quickly and catches many simple mistakes, but Vale is much, much more comprehensive and we can run this on our CI/CD.

I would appreciate any and all comments.

### CODE REVIEW

**Comment by akaszynski** [comment]
This PR is ready for review. If we are fine with using Vale, we will have several PRs for getting our docs in compliance and then one file one to enable the CI/CD. I don't want to end up with merge conflicts or long lived PRs.

**Comment by akaszynski** [comment]
@pyvista/developers, hoping on having some input on this, both in moving the source directory and in incorporating the use of Vale.sh

**Comment by adeak** [comment]
> This PR proposes the use of [Vale.sh](https://vale.sh/).

I haven't heard of it, so no objection from me :) As this seems to be a linter, I assume that using it (and reorganizing stuff to match) can't lead to any kind of lock-in.

>     * I suggest going with the Google Developer Documentation Style Guide as it seems to strike more of a conversational tone vs. some of the other style guides. I've included it in this PR.

I'll have to check what this guide says, just to educate myself. Do we have to vendor all those google stuff in the codebase for this to work? And will we have to keep updating all those vendored packages when google updates stuff?

>     * This PR moves the source of the documentation to the `source` directory. This seems more organized to me, especially with the inclusion of additional style configuration files.

I don't have strong feelings about this, but I'd consider saying something about this in the title of this PR. I had some bandwidth to look at things yesterday, but "Start using vale" was so unassuming that I didn't look at this.
 
>     * I'm using ChatGPT to help rewrite certain sections. Originally I used it to just get rid of first person, but I find that the rewritten sections read much better than the original sections. I'm honestly not the best technical writer, but I can discrimate between _ok_ and good writing, and I think we can achieve better results with this approach provided we do not lose the original content and meaning.

Firstly, I don't personally have anything against first person plural phrasing. I come from an academic background and "we" are the norm.

Secondly, by now when I see "ChatGPT" my heart skips a beat :smile: Knowing you and reading your description I'm sure you're using it right (insofar as using it can be right), i.e. using it to rephrase things and verifying the result. But I think it's really important to make sure that we don't converge to a state where later down the line a contributor starts taking output from ChatGPT (or any other machine learning system) for granted. If the committer doesn't observe due diligence, the burden of catching nonsense/fallacies is back on the reviewers. But now, instead of facing well-meaning errors stemming from mistaken/misspoken humans, the reviewers are facing the "malicious incompetence veiled by the elaborate phrasing of an intelligent psychopath" output that's typical of machine learning systems that don't actually have any domain knowledge. Any such help from ML systems must be subject to meticulous, sentence-by-sentence scrutiny from the person committing the text, with special care taken that the meaning is not subtly changed (i.e. unintented implications or unexpected weasel words/expressions being injected). It's not an exact fit to our domain (i.e. technical documentation), but anyway some of the critical answers [here](https://academia.stackexchange.com/questions/192013/is-it-ok-to-use-gpt-3-to-rewrite-your-own-paper-to-have-better-english) resonate with my views. I only mean all this as half a rant and half a warning to posterity :smile: 

>     * We can still use codespell since it runs quickly and catches many simple mistakes, but Vale is much, much more comprehensive and we can run this on our CI/CD.

Sounds good.

File: doc/source/user-guide/vtk_to_pyvista.rst
Comment (adeak) [inline doc/source/user-guide/vtk_to_pyvista.rst:93]: [According to the wiki](https://en.wikipedia.org/wiki/Fortran)

> The official language [standards](https://en.wikipedia.org/wiki/International_standard) for Fortran have referred to the language as "Fortran" with initial caps since Fortran 90.

Was this change here due to Google or due to ChatGPT? From the lack of changes I assume the former.

**Comment by tkoyama010** [comment]
+1 to this PR. 
I have three questions regarding the files under doc/styles:

1. What is the origin of these files?
2. Is the license of these files compatible with the PyVista MIT license?
3. How are these files updated when changes are made to the original files?

**Comment by tkoyama010** [comment]
If these files are your originals, then the above questions are unnecessary.

File: doc/source/user-guide/vtk_to_pyvista.rst
Comment (akaszynski) [reply doc/source/user-guide/vtk_to_pyvista.rst:93]: Google. My fault for the all caps.

**Comment by akaszynski** [comment]
> But I think it's really important to make sure that we don't converge to a state where later down the line a contributor starts taking output from ChatGPT (or any other machine learning system) for granted.

ChatGPT is great at generating output that is plausibly true, but without a domain expert reviewing it, it's not to be trusted at "face value." I'm fine if we use ChatGPT (or any large language model) to:

> "output[sic] a grammatically flawless version of the same text"

I'll review the different styles. Personally, I'm also fine with first person plural.

**Comment by akaszynski** [comment]
> +1 to this PR. I have three questions regarding the files under doc/styles:
> 
> 1. What is the origin of these files?
> 2. Is the license of these files compatible with the PyVista MIT license?
> 3. How are these files updated when changes are made to the original files?

Good point. These files come from https://github.com/errata-ai/packages, I'll be sure to include a link in a readme as well as the original license.

If we're going with first person plural pronouns, maybe the [Microsoft Writing Style Guide](https://docs.microsoft.com/en-us/style-guide/welcome/) might be a better choice.

**Comment by larsoner** [comment]
I have never heard of `vale` but my preference is usually to try to follow the general scientific Python trends when possible. To me that means using `black` nowadays. But if `vale` is compatible with that, then no objection from me...

**Comment by akaszynski** [comment]
I reset `vtk_to_pyvista.rst` and then reran with the Microsoft style guide. It's a bit more serious:
```
 source/user-guide/vtk_to_pyvista.rst
 9:18    error    'g. T' should have one space.   Microsoft.Spacing      
 44:26   error    Use 'that's' instead of 'that   Microsoft.Contractions 
                  is'.                                                   
 44:39   error    Punctuation should be inside    Microsoft.Quotes       
                  the quotes.                                            
 76:20   error    Use 'they're' instead of 'they  Microsoft.Contractions 
                  are'.                                                  
 77:9    error    Use 'doesn't' instead of 'does  Microsoft.Contractions 
                  not'.                                                  
 79:24   error    Don't spell out the number in   Microsoft.Units        
                  'like pixels'.                                         
 85:24   warning  Consider removing 'very'.       Microsoft.Adverbs      
 85:29   warning  Consider removing 'easily'.     Microsoft.Adverbs      
 101:63  warning  Consider removing 'nicely'.     Microsoft.Adverbs      
 103:16  error    Punctuation should be inside    Microsoft.Quotes       
                  the quotes.                                            
 155:9   warning  Consider removing 'heavily'.    Microsoft.Adverbs      
 156:14  error    's. F' should have one space.   Microsoft.Spacing      
 158:29  error    's. F' should have one space.   Microsoft.Spacing      
 190:39  warning  Consider using 'most' instead   Microsoft.Wordiness    
                  of 'almost all'.                                       
 208:63  error    'y. W' should have one space.   Microsoft.Spacing      
 238:71  error    Did you really mean             Vale.Spelling          
                  'datatypes'?                                           
 264:33  error    't. T' should have one space.   Microsoft.Spacing      
 274:64  error    's. F' should have one space.   Microsoft.Spacing      
 303:33  error    's. F' should have one space.   Microsoft.Spacing      
 306:10  error    Use 'it's' instead of 'it is'.  Microsoft.Contractions 
 308:46  error    'h. I' should have one space.   Microsoft.Spacing      
 308:50  warning  Consider using 'usually'        Microsoft.Wordiness    
                  instead of 'In most cases'.                            
 313:23  error    'a. F' should have one space.   Microsoft.Spacing      

✖ 17 errors, 6 warnings and 0 suggestions in 1 file.
```
I find that with a few tweaks, the Google style guide to be more aligned with our writing style, but with two tweaks:

```
- Removed `We.yml`. The developers at PyVista are fine with first person plural.
- Removed `Will.yml`. Future tense seems fine. We've tried implement this in PyAnsys and it's just not worth the effort to always speak in present tense. Maybe one day we'll revisit this.
```

**Comment by akaszynski** [comment]
> I have never heard of `vale` but my preference is usually to try to follow the general scientific Python trends when possible. To me that means using `black` nowadays. But if `vale` is compatible with that, then no objection from me...

If only we had an autoformatter for rst. Those that exist (like https://github.com/dzhu/rstfmt) I find wanting.

Furthermore, until we have AI that works really well (see the discussion above), documentation linting is the best we're going to get. At least if we can automate vale, we can reduce the number of comments @adeak has to make on my PRs :)

**Comment by akaszynski** [comment]
This is ready (again) for review. Changes to `vtk_to_pyvista.rst` are much less aggressive with zero use of ChatGPT.

**Comment by adeak** [comment]
I've pushed minor changes to make all files end with newlines.

The current diff (aside from the major relocation effort) seems really harmless. Thanks @akaszynski.

**Review by adeak — APPROVED** [review]

**Review by tkoyama010 — APPROVED** [review]
Let's give it a try! If we don't like it we could just Revert.

**Comment by akaszynski** [comment]
Waiting 24 hours before merge. Thanks all!

**Review by banesullivan — APPROVED** [review]
+100, merge at will! I'd like for us to rapidly get this working on CI and replace codespell with this.

**Pull request merged**
